### PR TITLE
Disable merchant

### LINKED ISSUE #31: User Story 42, Admin disables a merchant account
```
As an admin merchant
When I visit the merchant index page
And I click on a "disable" button for an enabled merchant
I am returned to the admin's merchant index page
And I see a flash message that the merchant's account is now disabled
And I see that the merchant's account is now disabled
This merchant cannot log in
```

DO THESE STORIES FIRST:
- User Story 41, Admin visits Merchant Index Page

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #35

child of #82

depends on #32

### PR DESCRIPTION
**Pull request by mikekarnes123**
- [] Check this if the PR has been approved by the team to be a quick patch and the rest of this form will not be filled out

# Test Coverage Percentage (Model/Feature):
- rspec - 100%
- rspec spec/models - x%

# What functionality does this accomplish?
Allows admins to disable merchants 

# What did you struggle on to complete?

# Implements/Fixes:
* description
closes #31 

# Testing Changes
- [] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [] My code has no unused/commented out code
- [] I have reviewed my code
-> I have commented my code, particularly in hard-to-understand areas
- [] I have fully tested my code

# Please include an emoji of how you feel about this branch:

### CODE REVIEW

File: spec/features/merchants/merchants_index_page_spec.rb [hunk 1]
```diff
@@ -70,5 +70,47 @@
...
+
+    it 'should enable/disable a merchant after corrisponding button has been clicked' do
+      admin = create(:admin)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(admin)
+
+      visit merchants_path
+      within "#merchant-#{@merchant_1.id}" do
+        click_link 'Disable'
+        expect(current_path).to eq(merchants_path)
+        #link should now be to enalbe to the account
```
Comment (milevy1) [inline spec/features/merchants/merchants_index_page_spec.rb]: enable*

File: spec/features/merchants/merchants_index_page_spec.rb [hunk 2]
```diff
@@ -70,5 +70,47 @@
...
+        expect(page).to have_link("Enable")
+        @merchant_1.reload
+        expect(@merchant_1.enabled).to eq(false)
+      end
+      #flash message isnt occuring within css tag
+      expect(page).to have_content("#{@merchant_1.name} is now disabled")
+
+      within "#merchant-#{@merchant_2.id}" do
```
Comment (milevy1) [inline spec/features/merchants/merchants_index_page_spec.rb]: What is line 90-99 testing that was not already tested in lines 79-88?

File: config/routes.rb [hunk 3]
```diff
@@ -15,6 +15,7 @@
   namespace :admin do
     get '/dashboard', to: 'users#dashboard', as: :dashboard
     resources :merchants, only: [:show, :update]
+    # patch '/merchants/:id/disable', to: 'merchants#update', as: :disable_merchant
```
Comment (milevy1) [inline config/routes.rb]: What's this commented out line for?

File: app/views/merchants/index.html.erb [hunk 4]
```diff
@@ -10,7 +10,7 @@
         <% end %></h3>
       <ul>
         <% if admin_user? %>
-          <li><%= link_to 'Disable' %></li>
+          <li><%= link_to 'Disable', admin_merchant_path(merchant, disable: true), method: :patch %></li>
```
Comment (milevy1) [inline app/views/merchants/index.html.erb]: Is this path confusing for anyone else? Would it simplify things to just hand-roll a "disable_merchant_path"

File: spec/features/merchants/merchants_index_page_spec.rb [hunk 5]
```diff
@@ -70,5 +70,47 @@
...
+
+  context 'as a merchant' do
+    describe 'attempts to login' do
+      it 'should prevent login if disabled' do
+        merchant = create(:inactive_merchant)
+        visit login_path
+        fill_in "email", with: merchant.email
+        fill_in "password", with: merchant.password
+        click_button "Log In"
+        expect(current_path).to eq(login_path)
```
Comment (milevy1) [inline spec/features/merchants/merchants_index_page_spec.rb]: I think it's working but is there any other way to expect that the login failed? Maybe with the flash message you made.

File: app/views/merchants/index.html.erb (hunk 4 above)
Comment (mikekarnes123) [reply app/views/merchants/index.html.erb]: @milevy1 Thats how I was doing it in my first attempt at the story, but me and Blake decided it was unrestful, we should decided as a group

File: config/routes.rb (hunk 3 above)
Comment (mikekarnes123) [reply config/routes.rb]: in case we wanted to switch back to a non restful

File: spec/features/merchants/merchants_index_page_spec.rb (hunk 1 above)
Comment (mikekarnes123) [reply spec/features/merchants/merchants_index_page_spec.rb]: YOU GUYS ARE SPELLING DICTATORS

File: spec/features/merchants/merchants_index_page_spec.rb (hunk 2 above)
Comment (mikekarnes123) [reply spec/features/merchants/merchants_index_page_spec.rb]: Nothing really, just figured there wasn't really a reason to not test it

File: spec/features/merchants/merchants_index_page_spec.rb (hunk 5 above)
Comment (mikekarnes123) [reply spec/features/merchants/merchants_index_page_spec.rb]: User story didn't ask for a flash, but its not a bad idea

File: config/routes.rb (hunk 3 above)
Comment (milevy1) [reply config/routes.rb]: Gotcha

File: spec/features/merchants/merchants_index_page_spec.rb (hunk 5 above)
Comment (milevy1) [reply spec/features/merchants/merchants_index_page_spec.rb]: Well the flash already exists:  `flash[:error] = 'Your credentials are incorrect.'`

File: app/controllers/admin/merchants_controller.rb [hunk 6]
```diff
@@ -6,10 +6,18 @@ def show
   end
 
   def update
-    user_to_be_updated = User.find(params[:id])
-    user_to_be_updated.disable_merchant_items
-    user_to_be_updated.update(role: 0)
-    flash[:notice] = "Merchant #{user_to_be_updated.name} downgraded"
-    redirect_to admin_user_path(user_to_be_updated)
+    if params[:disable]
+      merchant = User.find(params[:id])
```
Comment (milevy1) [inline app/controllers/admin/merchants_controller.rb]: To dry this up, can we move line 10 above the if statement.  We could then delete line 16

File: app/controllers/admin/merchants_controller.rb [hunk 7]
```diff
@@ -6,10 +6,18 @@ def show
...
+      merchant.disable_merchant_items
+      merchant.update(enabled: false)
+      flash[:merchant_disabled] = "#{merchant.name} is now disabled"
+      redirect_to merchants_path
+    else
```
Comment (milevy1) [inline app/controllers/admin/merchants_controller.rb]: Can we make this an elsif so it's clear that this path is for downgrading a merchant to a user

Comment (mikekarnes123) [reply app/controllers/admin/merchants_controller.rb]: Done!
